src/pages/workspaces/workspace/analysis/tool-utils.ts

```diff
@@ -46,41 +46,68 @@ export interface AppTool extends Tool {
   appType: string
 }
 
+const RStudio: RuntimeTool = { label: toolLabels.RStudio, ext: ['Rmd', 'R'] as Extension[], imageIds: ['RStudio'], defaultImageId: 'RStudio', defaultExt: 'Rmd' as Extension }
+
+const Jupyter: RuntimeTool = {
+  label: toolLabels.Jupyter,
+  ext: ['ipynb' as Extension],
+  isNotebook: true,
+  imageIds: ['terra-jupyter-bioconductor', 'terra-jupyter-bioconductor_legacy', 'terra-jupyter-hail', 'terra-jupyter-python', 'terra-jupyter-gatk', 'Pegasus', 'terra-jupyter-gatk_legacy'],
+  defaultImageId: 'terra-jupyter-gatk',
+  isLaunchUnsupported: true,
+  defaultExt: 'ipynb' as Extension
+}
+
+const JupyterLab: RuntimeTool = {
+  label: toolLabels.JupyterLab,
+  isNotebook: true,
+  ext: ['ipynb' as Extension],
+  isLaunchUnsupported: false,
+  defaultExt: 'ipynb' as Extension,
+  imageIds: [],
+  defaultImageId: '',
+}
+
+const Galaxy: AppTool = { label: toolLabels.Galaxy, appType: 'GALAXY' }
+
+const Cromwell: AppTool = { label: toolLabels.Cromwell, appType: 'CROMWELL', isHidden: !isCromwellAppVisible(), isPauseUnsupported: true }
+
 export const appTools: Record<AppToolLabel, AppTool> = {
-  Galaxy: { label: 'Galaxy', appType: 'GALAXY' },
-  Cromwell: { label: 'Cromwell', appType: 'CROMWELL', isHidden: !isCromwellAppVisible(), isPauseUnsupported: true }
+  Galaxy,
+  Cromwell
 }
 
 export const runtimeTools: Record<RuntimeToolLabel, RuntimeTool> = {
-  RStudio: { label: 'RStudio', ext: ['Rmd', 'R'] as Extension[], imageIds: ['RStudio'], defaultImageId: 'RStudio', defaultExt: 'Rmd' as Extension },
-  Jupyter: {
-    label: 'Jupyter',
-    ext: ['ipynb' as Extension],
-    isNotebook: true,
-    imageIds: ['terra-jupyter-bioconductor', 'terra-jupyter-bioconductor_legacy', 'terra-jupyter-hail', 'terra-jupyter-python', 'terra-jupyter-gatk', 'Pegasus', 'terra-jupyter-gatk_legacy'],
-    defaultImageId: 'terra-jupyter-gatk',
-    isLaunchUnsupported: true,
-    defaultExt: 'ipynb' as Extension
-  },
+  RStudio,
+  Jupyter,
   // azure should be changed in backend to be a same as jupyter
   // we should use analysis file's cloud provider to determine azure/other logic for jupyter
-  Azure: {
-    label: 'Azure',
-    isNotebook: true,
-    ext: ['ipynb' as Extension],
-    isAzureCompatible: true,
-    isLaunchUnsupported: false,
-    defaultExt: 'ipynb' as Extension,
-    imageIds: [],
-    defaultImageId: ''
-  }
+  JupyterLab,
 }
 
 export const tools: Record<ToolLabel, Tool> = {
   ...runtimeTools,
   ...appTools,
-  terminal: { label: 'terminal' },
-  spark: { label: 'spark' }
+  terminal: { label: toolLabels.terminal },
+  spark: { label: toolLabels.spark }
+}
+
+export const cloudRuntimeTools: Record<CloudProviderType, Partial<Record<ToolLabel, RuntimeTool>>> = {
```

Comment:
Hmm. It seems this would be a bit simpler if it was typed a little different. In the places I see it used there's a _.values() on the object and treat it's as an array anyways. 
```
export const cloudRuntimeTools: Record<CloudProviderType, RuntimeTool[]> = {
GCP: [RStudio, Jupyter],
AZURE: [JupyterLab]
}
```